include/dgl/aten/coo.h

```diff
@@ -64,9 +64,12 @@ struct COOMatrix {
         data(darr),
         row_sorted(rsorted),
         col_sorted(csorted) {
-    is_pinned = (aten::IsNullArray(row) || row.IsPinned()) &&
-                (aten::IsNullArray(col) || col.IsPinned()) &&
-                (aten::IsNullArray(data) || data.IsPinned());
+    if (!aten::IsNullArray(row) || !aten::IsNullArray(col) ||
```

Comment:
how about wrapping these `IsNullArray()` check into a member function of `COOMatrix`: `IsEmpty()`to remove code redundancy?